resources/aws/bucket.go

```diff
@@ -50,3 +51,42 @@ func (b *Bucket) Delete() error {
 
 	return nil
 }
+
+type BucketObject struct {
+	Name   string
+	Data   string
+	Bucket *Bucket
+	AWSEntity
+}
+
+func (bo *BucketObject) CreateIfNotExists() (bool, error) {
+	return false, fmt.Errorf("not impelented")
+}
+
+func (bo *BucketObject) CreateOrFail() error {
+	if bo.Bucket == nil {
+		return fmt.Errorf("Object needs to belong to some bucket")
```

Comment:
I believe we want to use `microerror` here too, cfr. https://github.com/giantswarm/aws-operator/pull/86#discussion_r108380977